/*
 * Copyright (c) 2024 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto2";

option optimize_for = LITE_RUNTIME;
package i18n.phonenumbers;

message GeocodingInfo {
    repeated PrefixesInfo prefixes_info = 1;
    repeated string languages = 2;
    required LanguageCodeInfo language_code_info = 3;
    repeated CountriesInfo countries_info = 4;
    repeated int32 countries = 5;
    required CountryCodeInfo country_code_info = 6;
}

message PrefixesInfo {
    required int32 prefixes_num = 1;
    repeated int32 prefixes = 2;
    repeated string descriptions = 3;
    required int32 lengths_num = 4;
    repeated int32 lengths = 5;
}

message LanguageCodeInfo {
    required int32 language_codes_num = 1;
    repeated string language_codes = 2;                                     
}

message CountriesInfo {
    required int32 country_languages_num = 1;
    repeated string country_languages = 2;
}

message CountryCodeInfo {
    required int32 country_codes_num = 1;
    repeated int32 country_codes = 2;
}
